### PR TITLE
Dependencies - update terraform-plugin-framework minor version

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description

Updates
- terraform-plugin-framework package to v1.5.0
- terraform-plugin-go to v0.21.0
- terraform-plugin-testing to v1.6.0
- terraform-plugin-sdk/v2 v2.30.0

Hopefully this will make dependabot happy. We've had at least 4 PRs of package updates, fail to compile recently due to code in dependencies. #355, #341, #331, #330, #321, #391, #382

Updated the README to use 1.6 while I'm here, the framework package has updates to improve performance with 1.6.

The test disabled in [f3bc51e](https://github.com/juju/terraform-provider-juju/pull/359/commits/f3bc51ea56d835b25fcaa10a43eb4ae186e3d28a) will be fixed in a follow on PR. It's an existing issue and not a regression.

## Type of change

- Maintenance work (repository related, like Github actions, or revving the Go version, etc.)

## Additional Info

JUJU-5182

### CODE REVIEW

**Comment by hmlanigan** [comment]
This change of the framework pkg version from 1.3.x to 1.4.x causes test failures. Holding off until after 0.10.1 get's released.

File: internal/juju/machines.go [hunk 1]
```diff
@@ -252,21 +297,49 @@ func (c machinesClient) ReadMachine(input ReadMachineInput) (ReadMachineResponse
...
+			if attempt%4 == 0 {
+				message := fmt.Sprintf("waiting for machine %q", input.ID)
+				if attempt != 4 {
+					message = "still " + message
+				}
+				c.Debugf(message)
+			}
+		},
+		BackoffFunc: retry.DoubleDelay,
+		Attempts:    30,
```
Comment (cderici) [inline internal/juju/machines.go:335]: is 30 attempts a little too many?

**Review by cderici — APPROVED** [review]
LGTM 👍

File: internal/juju/machines.go (hunk 1 above)
Comment (hmlanigan) [reply internal/juju/machines.go:335]: 30 is arbitrary for sure. In practicality, it's a max time of 61 sec due to the backoff func of DoubleDelay and wait time of 1 sec. 

We also have the ctx.Done() that's passed through from the terraform client as a backstop.

**Pull request merged**